src/client.ts

```
@@ -94,11 +105,7 @@ export class APIClient {
       return response.json();
     } catch (err) {
       this.logger?.warn(err, 'Could not get Request on API Client');
-      throw new IntegrationProviderAPIError({
-        endpoint: endpoint,
-        status: err.status,
-        statusText: err.statusText,
-      });
+      throw err;
```

This could actually be considered a bug in our old code - when we make the request above, we throw certain errors depending on the scenario (i.e. retryable vs non-retryable, and now auth errors for 401s). However, this catch block completely disregarded the error we threw when making the request, and constructed a new IntegrationProviderAPIError. This means we lost any "retryable" context that we may have had on the error. 

We should instead just rethrow the error we threw when making the request.